File: Editrans.py
import urllib.request
import os

#Variables
URLEditrans="https://www2.agenciatributaria.gob.es/L/inwinvoc/es.aeat.dit.adu.adht.editran.NumRefEditran?mod=347"
desktop = os.path.join(os.path.join(os.environ['USERPROFILE']), 'Desktop')
numberOfEditrans= 2
textToSearch = 'Justificante: <strong>'

#Deleting previous file and creating a new one
if os.path.exists(desktop +"//editrans.txt"):
    os.remove(desktop +"//editrans.txt")
f = open(desktop +"//editrans.txt","a")

#Get number and write it on the file
for x in range(numberOfEditrans):
    response = urllib.request.urlopen(URLEditrans)
    the_page = response.read().decode('latin-1')
    codeEditrans = the_page[the_page.find(textToSearch) + 22:the_page.find(textToSearch) + 36].replace(" ", "")

    if x == numberOfEditrans-1:
        f.write(codeEditrans)
    else:
        f.write(codeEditrans + "\n")
f.close()

File: EdiTrans.java
package Editrans;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.net.URL;
import java.net.URLConnection;

public class EdiTrans {


	public static void main(String[] args) throws IOException {
		
		//Declaración de las variables a utilizar en el programa
		int numero_justificantes=150;
		String palabra_a_buscar="Justificante: <strong>";
		URL test = new URL("https://www2.agenciatributaria.gob.es/L/inwinvoc/es.aeat.dit.adu.adht.editran.NumRefEditran?mod=347");
		String inputLine;
		
		//Obtenemos el código fuente de la página, y mediante una expresión regular nos quedaremos con los dígitos necesarios del Editrans
		for (int i=1; i<=numero_justificantes;i++) {
		URLConnection uc = test.openConnection();
		uc.addRequestProperty("User-Agent", "Mozilla/4.0");
		BufferedReader in = new BufferedReader(new InputStreamReader(uc
				.getInputStream()));
		while ((inputLine = in.readLine()) != null) {
				if (inputLine.contains(palabra_a_buscar)){
					String numero =inputLine.substring(inputLine.indexOf("g>")+2,inputLine.indexOf("g>")+16).replace(" ","");
					System.out.println(numero);
				}
		}

		in.close();

		}


	}

}
